gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ConsultationListMapper.java

```diff
@@ -127,13 +136,21 @@ public ListResource mapToCategory(ListResource topic, RCMRMT030101UKCompoundStat
         return category;
     }
 
-    private DateTimeType getDate(RCMRMT030101UKCompoundStatement compoundStatement, ListResource parentList) {
+    private DateTimeType getDate(RCMRMT030101UKCompoundStatement compoundStatement, ListResource parentList,
+                                 RCMRMT030101UK04EhrExtract ehrExtract) {
         if (compoundStatement != null && compoundStatement.getAvailabilityTime() != null
             && compoundStatement.getAvailabilityTime().getValue() != null) {
             return DateFormatUtil.parseToDateTimeType(compoundStatement.getAvailabilityTime().getValue());
+        } else if (parentList.getDateElement() != null) {
+            return parentList.getDateElement();
+        } else {
+            if (ehrExtract.getComponent().get(0) != null && ehrExtract.getComponent().get(0).getEhrFolder().getComponent().get(0) != null) {
```

Comment:
This approach of getting the EhrComposition won't work. This just gets the very first `EhrComposition` within the payload, which might not be the one we want to look at.

Reply:
ok will look at a better solution.